pkgs/development/python-modules/aria2p/default.nix

```diff
@@ -0,0 +1,23 @@
+{ buildPythonPackage, fetchPypi, lib, pythonOlder
+, asciimatics, loguru, requests, setuptools, websocket_client
+}:
+
+buildPythonPackage rec {
+  pname = "aria2p";
+  version = "0.7.0";
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0aaricdj4kvd08m5a8f1bjlcndfdsac3mc0wain8fg0m2djsx2pq";
+  };
```

Comment:
pypi doesn't distribute tests, please checkout from github

```
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
/nix/store/lx6h0y623g9010fhzj1k1kq1y5jgikqn-python3.7-aria2p-0.7.0
```

Reply:
Had to change many things to comply with this requirement:
- `aria2p` relies on `poetry` to build from source
- `poetry` silently creates a virtualenv inside `$HOME`, which is forbidden with `nix`; it is possible to change the virtualenv location (cf https://python-poetry.org/docs/configuration/) but this is only available from versions > 1.0
- current `nixpkgs` has an old version of `poetry`; upgrading it requires upgrading other tools, namely: `cachy`, `clikit`, `importlib-metadata`, `keyring`

Reply:
Reverted `poetry` changes, using `export HOME=$TMPDIR` instead. Pull-request is much simpler and should work now.